src/function/cast/string_cast.cpp

```diff
@@ -166,9 +167,73 @@ bool StringListCastLoop(string_t *source_data, ValidityMask &source_mask, Vector
 	       all_converted;
 }
 
-bool StringListCast(Vector &source, Vector &result, idx_t count, CastParameters &parameters) {
+static LogicalType InitVarcharStructType(const LogicalType &target) {
+	return LogicalType::STRUCT(
+	    child_list_t<LogicalType>(StructType::GetChildCount(target), make_pair("", LogicalType::VARCHAR)));
+}
+
+bool VectorStringToStruct::StringToNestedTypeCastLoop(string_t *source_data, ValidityMask &source_mask, Vector &result,
+                                                      ValidityMask &result_mask, idx_t count,
+                                                      CastParameters &parameters, const SelectionVector *sel) {
+
+	auto varchar_struct_type = InitVarcharStructType(result.GetType());
+	Vector varchar_vector(varchar_struct_type, count);
+	auto &child_vectors = StructVector::GetEntries(varchar_vector);
+	auto &result_children = StructVector::GetEntries(result);
+
+	string_map_t<idx_t> child_names;
+	vector<ValidityMask *> child_masks;
+	// to cover missing values, set all child masks to invalid, then only set the found values to valid
+	for (idx_t child_idx = 0; child_idx < result_children.size(); child_idx++) {
+		child_names.insert({StructType::GetChildName(result.GetType(), child_idx), child_idx});
+		if (result.GetVectorType() == VectorType::CONSTANT_VECTOR) {
+			(child_vectors[child_idx])->SetVectorType(VectorType::CONSTANT_VECTOR);
+			child_masks.emplace_back(&ConstantVector::Validity(*child_vectors[child_idx]));
+		} else {
+			child_masks.emplace_back(&FlatVector::Validity(*child_vectors[child_idx]));
+		}
+		child_masks[child_idx]->SetAllInvalid(count);
+	}
+
+	bool all_converted = true;
```

Comment:
nit: can we move this to below this loop? It's not altered in the loop below, HandleVectorCastError takes it as an argument but it's always true

Reply:
all_converted is passed by reference and set to false, so I need the variable

Reply:
Ah that makes perfect sense, I should have checked that before commenting